features/events/daemon/src/main/java/org/opennms/netmgt/eventd/AbstractEventUtil.java

```diff
@@ -371,239 +369,19 @@ public static String escape(String inStr, char delimchar) {
 		return outBuffer.toString();
 	}
 
-	/**
-	 * Get the value of the parm for the event
-	 *
-	 * @param parm
-	 *            the parm for which value is needed from the event
-	 * @param event
-	 *            the event whose parm value is required
-	 * @return value of the event parm/element
-	 */
-	@Override
-	public String getValueOfParm(String parm, Event event) {
+	@Autowired
+	private TransactionOperations transactionOperations;
 
-		String retParmVal = null;
-		final String ifString = event.getInterface();
-		
-		if (parm.equals(TAG_UEI)) {
-			retParmVal = event.getUei();
-		}
-		if (parm.equals(TAG_EVENT_DB_ID)) {
-			if (event.hasDbid()) {
-				retParmVal = Integer.toString(event.getDbid());
-			} else {
-				retParmVal = "eventid-unknown";
-			}
-		} else if (parm.equals(TAG_SOURCE)) {
-			retParmVal = event.getSource();
-		} else if (parm.equals(TAG_DPNAME)) {
-			retParmVal = event.getDistPoller();
-		} else if (parm.equals(TAG_DESCR)) {
-			retParmVal = event.getDescr();
-				} else if (parm.equals(TAG_LOGMSG)) {
-					retParmVal = event.getLogmsg().getContent();
-		} else if (parm.equals(TAG_NODEID)) {
-			retParmVal = Long.toString(event.getNodeid());
-		} else if (parm.equals(TAG_NODELABEL)) {
-			retParmVal = Long.toString(event.getNodeid());
-			String nodeLabel = null;
-			if (event.getNodeid() > 0) {
-				try {
-					nodeLabel = getNodeLabel(event.getNodeid());
-				} catch (SQLException e) {
-					// do nothing
-				}
-			}
-			if (nodeLabel != null)
-				retParmVal = WebSecurityUtils.sanitizeString(nodeLabel);
-			else
-				retParmVal = "Unknown";
-		} else if (parm.equals(TAG_FOREIGNSOURCE)) {
-			retParmVal = "";
-			if (event.getNodeid() > 0) {
-				try {
-					String foreignSource = getForeignSource(event.getNodeid());
-					if (foreignSource != null) {
-						retParmVal = WebSecurityUtils.sanitizeString(foreignSource);
-					}
-				} catch (SQLException e) {
-					// do nothing
+	private final LoadingCache<String, EventTemplate> eventTemplateCache = CacheBuilder.newBuilder()
```

Comment:
Multiple instance of this bean are created in the runtime (i.e. https://github.com/OpenNMS/opennms/blob/develop/opennms-services/src/main/resources/META-INF/opennms/applicationContext-notifd.xml#L65) so each of these will end up with their own cache.

Can we find a way to de-duplicate the caches, or ideally, de-duplicate these beans?